javaslang-test/src/main/java/javaslang/test/Gen.java

```diff
@@ -277,6 +293,17 @@
         return choose(1, size).flatMap(n -> new Frequency().gen(n, stream));
     }
 
+  /**
+   * Intersperse values from this generator instance with those of another.
+   *
+   * @param other another T generator to accept values from.
+   * @return A new T generator
+   */
+    default Gen<T> intersperse(Gen<T> other) {
+        AtomicReference<Tuple2<Gen<T>, Gen<T>>> interspersedReference = new AtomicReference<>(Tuple.of(other, this));
+        return random -> interspersedReference.updateAndGet(generators -> Tuple.of(generators._2, generators._1))._1.apply(random);
+    }
```

Comment:
Your questions of the prev PR:
- We do not have an in-place swap. This is not possible for Tuple2 because of the generic types T1 != T2 (wlog)
- Lets implement it as iterator with a boolean `switch = !switch` state to avoid new Tuple instances:

```java
Iterator<Gen<T>> iter = new Java.util.Iterator<>() {
    boolean switch = false;
    hasNext() = true
    next() = {
        gen = switch ? ... : ...;
        seitch = !switch;
        return gen;
    }
}
return rnd -> iter.next().apply(rnd);
```